src/golang/lib/models/shared/integration_config.go

```diff
@@ -100,9 +139,58 @@ type SlackConfig struct {
 	Enabled  bool              `json:"enabled"`
 }
 
+type DynamicK8sConfig struct {
+	Keepalive   string `json:"keepalive"`
+	CpuNodeType string `json:"cpu_node_type"`
+	GpuNodeType string `json:"gpu_node_type"`
+	MinCpuNode  string `json:"min_cpu_node"`
+	MaxCpuNode  string `json:"max_cpu_node"`
+	MinGpuNode  string `json:"min_gpu_node"`
+	MaxGpuNode  string `json:"max_gpu_node"`
+}
+
+func (config *DynamicK8sConfig) ToMap() map[string]string {
+	configMap := make(map[string]string)
+
+	valueOf := reflect.ValueOf(config).Elem()
+	typeOf := valueOf.Type()
+
+	for i := 0; i < valueOf.NumField(); i++ {
+		field := valueOf.Field(i)
+		fieldName := typeOf.Field(i).Tag.Get("json")
+		fieldValue := fmt.Sprintf("%v", field.Interface())
+
+		if fieldValue != "" {
+			configMap[fieldName] = fieldValue
+		}
+	}
+
+	return configMap
+}
+
+func (config *DynamicK8sConfig) Merge(newConfig *DynamicK8sConfig) {
```

Comment:
Could we add docstrings for these two methods? Especially since their implementations are a little opaque.

Reply:
I also wonder if `Update()` is a better term? Don't have a big preference though